### DISCUSSION TITLE
Can't get GraphQL where filter to work with nested fields

### DISCUSSION THREAD

**Discussion by monkeymosher** [strapi/strapi#6131] [Legacy Topics]
Hi all!

I've been struggling with a simple query for quite a while now and thought you might help me figure out what I'm doing wrong.

I have a simple model: music records that are associated with artists. One record can have many artists and one artist can have many records.

What I'm trying to achieve here is to filter all the records for one specific band. 
My query goes as follow:
```
query Records($limit: Int!, $sort: String!) {
  records(limit: $limit, sort: $sort) {
    id
    title
    bands(sort: "name:desc",where: {name:"Descendents"}) {
      id
      name
      records{
        id
      }
    }
  }
}
```

In the results I see that I get my entire list of records unfiltered, while the bands in each record does apply the filter correctly. 

**My question is: how can I add my "band where" at the "records" level?**

Thanks!
![Screen Shot 2020-05-08 at 2 19 59 PM](https://user-images.githubusercontent.com/4013355/81436314-48dea100-9137-11ea-9a99-119e735e34f4.png)

**Comment by derrickmehaffy** [comment]
Are you using SQLite?

**Reply by monkeymosher** [reply]
Yes sir :) 

I also tried putting my where at the records level with `where: { bands.name:"Descendents"}` but that doesn't work either.

**Reply by derrickmehaffy** [reply]
Indeed there are a number of filtering options that don't work with SQLite, I'd suggest testing on an actual database.

**Reply by monkeymosher** [reply]
Any idea why it works when I call the api via `http://localhost:1337/records/?bands.name_contains=descendents ` and not via GraphQL?
![Screen Shot 2020-05-14 at 3 42 28 PM](https://user-images.githubusercontent.com/4013355/81978720-ed5e5880-95f9-11ea-94b7-2d309ec0c705.png)

**Comment by monkeymosher** [comment]
After struggling, I finally found the solution. My original query was wrongly structured. Here is how I made work, it if can help anyone else.

```
query Records($limit: Int!, $sort: String!) {
  records(limit: $limit, sort: $sort, where: { bands: {name_contains:"descendents"}}) {
    id
    title
    is_colour
    is_handnumbered
    is_testpress
    format
    location
    bands(sort: "name:desc") {
      id
      name
      records{
        id
      }
    }
  }
}
```
